### PR TITLE
added cob4-3

### PR DESCRIPTION
**Pull request by NadiaHG**

### CODE REVIEW

File: cob_hardware_config/test/urdf_cob4-3.test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<launch>
+
+  <test test-name="cob4_3_test" pkg="cob_description" type="test_urdf.py" args="$(find cob_hardware_config)/cob4-3/urdf/cob4-6.urdf.xacro"/>
```
Comment (floweisshardt) [inline cob_hardware_config/test/urdf_cob4-3.test]: should be cob4-3.urdf.xacro

File: cob_hardware_config/cob4-4/config/base_driver.yaml [hunk 2]
```diff
@@ -45,4 +45,3 @@ nodes:
   fr_caster_r_wheel_joint:
     <<: *drive_wheel
     id: 6
-
```
Comment (fmessmer) [inline cob_hardware_config/cob4-4/config/base_driver.yaml:48]: undo this change as it is in a cob4-4 file

File: cob_hardware_config/cob4-3/urdf/cob4-3.urdf.xacro [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro"
+       name="cob4-6" >
```
Comment (fmessmer) [inline cob_hardware_config/cob4-3/urdf/cob4-3.urdf.xacro]: cob4-3

File: cob_hardware_config/cob4-3/config/teleop.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#cob4-4
```
Comment (fmessmer) [inline cob_hardware_config/cob4-3/config/teleop.yaml]: cob4-3

File: cob_controller_configuration_gazebo/CMakeLists.txt [hunk 5]
```diff
@@ -10,6 +10,8 @@ set(robotlist
   cob3-9
   cob4-1
   cob4-2
+  cob4-3
+  cob4-4
```
Comment (fmessmer) [inline cob_controller_configuration_gazebo/CMakeLists.txt:14]: cob4-4 should not be added in this PR

File: cob_bringup/CMakeLists.txt [hunk 6]
```diff
@@ -11,6 +11,8 @@ set(robotlist
   cob3-9
   cob4-1
   cob4-2
+  cob4-3
+  cob4-4
```
Comment (fmessmer) [inline cob_bringup/CMakeLists.txt:15]: cob4-4 should not be added in this PR

**Comment by fmessmer** [comment]
We should not forget to add the new scan_unifier introduced in https://github.com/ipa320/cob_robots/pull/364
(i.e. merge this PR after #364 and add launch and yaml file for scan_unifier to this PR)

File: cob_bringup/robots/cob4-3.xml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+		</include>
+		<include file="$(find cob_script_server)/launch/script_server.launch" />
+		<!--include file="$(find cob_bringup)/tools/base_collision_observer.launch" >
+			<arg name="robot" value="$(arg robot)" />
+		</include-->
+
+		<!-- aggregated robot_state_publisher -->
+		<include file="$(find cob_bringup)/tools/robot_state_publisher.launch">
+			<arg name="robot" value="$(arg robot)" />
+		</include>
```
Comment (floweisshardt) [inline cob_bringup/robots/cob4-3.xml:67]: add scan_unifier (see other robots)

File: cob_bringup/CMakeLists.txt (hunk 6 above)
Comment (benmaidel) [inline cob_bringup/CMakeLists.txt:15]: or rename your PR :wink:

File: cob_controller_configuration_gazebo/CMakeLists.txt (hunk 5 above)
Comment (fmessmer) [inline cob_controller_configuration_gazebo/CMakeLists.txt:14]: I changed my mind!
I think it is better to fix something whenever a bug is detected!

File: cob_bringup/CMakeLists.txt (hunk 6 above)
Comment (fmessmer) [inline cob_bringup/CMakeLists.txt:15]: same as above....changed my mind....it's better to fix something than to be picky about such small things....Thanks for detecting it!

File: cob_controller_configuration_gazebo/launch/robots/default_controllers_cob4-3.launch [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+	<include file="$(find cob_bringup)/drivers/laser_scan_filter.launch">
+		<arg name="robot" value="$(arg robot)"/>
+		<arg name="name" value="left"/>
+	</include>
+	<include file="$(find cob_bringup)/drivers/laser_scan_filter.launch">
+		<arg name="robot" value="$(arg robot)"/>
+		<arg name="name" value="right"/>
+	</include>
+
+  <include file="$(find cob_bringup)/drivers/relayboard.launch">
```
Comment (fmessmer) [inline cob_controller_configuration_gazebo/launch/robots/default_controllers_cob4-3.launch]: minor indentation issue

File: cob_hardware_config/cob4-3/config/diagnostics_analyzers.yaml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+## remove diagnostics from gazebo in simulated case
+  simulation:
+    type: diagnostic_aggregator/DiscardAnalyzer
+    path: none
+    contains: 'joint'
+  simulation2:
+    type: diagnostic_aggregator/DiscardAnalyzer
+    path: none
+    contains: 'Controller'
```
Comment (fmessmer) [inline cob_hardware_config/cob4-3/config/diagnostics_analyzers.yaml:71]: I guess after the simulated diagnostics have been removed in #358, these entries could be removed for all robots, right?

Comment (NadiaHG) [inline cob_hardware_config/cob4-3/config/diagnostics_analyzers.yaml:71]: I will open a new pull request removing these lines for all the robots, as soon as this PR is merged.

Comment (fmessmer) [inline cob_hardware_config/cob4-3/config/diagnostics_analyzers.yaml:71]: New issue for this: https://github.com/ipa320/cob_robots/issues/380
